src/widgets/barchart/mod.rs

```diff
@@ -196,6 +205,71 @@ impl<'a> BarChart<'a> {
         }
     }
 
+    fn render_horizontal_bars(self, buf: &mut Buffer, bars_area: Rect, max: u64) {
+        // convert the bar values to ratatui::symbols::bar::Set
+        let groups: Vec<Vec<u16>> = self
+            .data
+            .iter()
+            .map(|group| {
+                group
+                    .bars
+                    .iter()
+                    .map(|bar| (bar.value * u64::from(bars_area.width) / max) as u16)
+                    .collect()
+            })
+            .collect();
+
+        // print all visible bars (without labels and values)
+        let mut bar_y = bars_area.bottom() - 1;
+        for (group_data, mut group) in groups.into_iter().zip(self.data) {
+            let bars = std::mem::take(&mut group.bars);
+
+            let label_offset = bars.len() as u16 * (self.bar_width + self.bar_gap) + self.bar_gap;
+            // if group_gap is zero, then there is no place to print the group label
+            // check also if the group label is still inside the visible area
+            if self.group_gap > 0 && bar_y >= bars_area.top() + label_offset {
+                let label_rect = Rect {
+                    y: bar_y - label_offset,
+                    ..bars_area
+                };
+                group.render_label(buf, label_rect, self.label_style);
+            }
+
+            for (bar_size, bar) in group_data.into_iter().zip(bars) {
+                let bar_style = bar.style.patch(self.bar_style);
+
+                for y in 0..self.bar_width {
+                    for x in 0..bar_size {
+                        buf.get_mut(bars_area.left() + x, bar_y - y)
+                            .set_symbol(self.bar_set.full)
+                            .set_style(bar_style);
+                    }
+                    for x in bar_size..bars_area.width {
+                        buf.get_mut(bars_area.left() + x, bar_y - y)
+                            .set_symbol(self.bar_set.empty)
+                            .set_style(bar_style);
+                    }
+                }
```

Comment:
I suspect that this sort of code calls for a nicer method on buffer to set an entire Rect perhaps?
This makes it a bit more obvious how this works:

```suggestion
                for y in 0..self.bar_width {
                    for x in 0..bars_area.width {
                        let symbol = if x < bar_size { self.bar_set.full } else { self.bar_set.empty };
                        buf.get_mut(bars_area.left() + x, bar_y - y)
                            .set_symbol(symbol)
                            .set_style(bar_style);
                    }
                }
```